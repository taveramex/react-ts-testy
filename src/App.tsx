import MyCoolApp from "./components/MyCoolApp"

function App() {

  return (
    <>
      <h1>Test App Component</h1>
      <div>This is a Test App to create stuff and learn</div>
      <MyCoolApp  />
    </>
  )
}

export default App
