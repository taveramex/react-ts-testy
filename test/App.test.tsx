import { describe, expect,test } from "vitest"
import { screen, render } from "@testing-library/react"
import App from "../src/App";

describe("App", ()=>{
    test("data-testid field", ()=>{
        render(<App />);
        const dataTest = screen.getByTestId("testingTest");
        expect(dataTest.innerHTML).toContain('giberish');
    });
    test('match whole component Snapshot',()=>{
        const {container} = render(<App />);
        expect(container.innerHTML).toMatchSnapshot();
    });
    test('match dataTestid Snapshot',()=>{
        render(<App />);
        expect(screen.getByTestId("testingTest")).toMatchSnapshot();
    });
});