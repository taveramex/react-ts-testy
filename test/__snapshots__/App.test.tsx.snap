// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > match dataTestid Snapshot 1`] = `
<div
  data-testid="testingTest"
>
  just some test giberish
</div>
`;

exports[`App > match whole component Snapshot 1`] = `"<h1>Test App Component</h1><div>This is a Test App to create stuff and learn</div><div data-testid="testingTest">just some test giberish</div><span>not cool App</span><p style="color: red; background-color: rgb(250, 250, 250); border-radius: 0;">this is awful to format style</p><p style="background-color: red; color: green; margin-top: 30px;">other aberration to see but a little less awful</p>"`;
